server/server.go

```diff
@@ -122,6 +127,9 @@ type Server struct {
 	idTokensValidFor time.Duration
 
 	logger logrus.FieldLogger
+
+	// mutex for the connectors map.
+	mu sync.Mutex
```

Comment:
nit: It's generally good style to put the mutex above what it's protecting. E.g.

```
type foo struct {
   a int

    mu sync.Mutext // protects b
    b int

    c int
}
```